cpp/src/copying/get_element.cu

```diff
@@ -0,0 +1,140 @@
+
+    using ScalarType = cudf::experimental::scalar_type_t<T>;
+    auto typed_s = static_cast<ScalarType *>(s.get());
+    
+    auto device_s = get_scalar_device_view(*typed_s);
+    auto device_col = column_device_view::create(input, stream);
+    
+    thrust::for_each_n(rmm::exec_policy(stream)->on(stream), 
+                       thrust::make_constant_iterator(0), 1, 
+      [ device_s, d_col=*device_col, index ] __device__ (auto) mutable {
+        device_s.set_value(d_col.element<T>(index));
+        device_s.set_valid(d_col.is_valid(index));
+      });
+    return s;
+  }
+
+  template<typename T>
+  std::enable_if_t<std::is_same<T, string_view>::value, 
+  std::unique_ptr<scalar>> operator()(
+    column_view const& input,
+    size_type index,
+    cudaStream_t stream = 0,
+    rmm::mr::device_memory_resource *mr = rmm::mr::get_default_resource())
+  {
+    auto device_col = column_device_view::create(input, stream);
+
+    rmm::device_scalar<string_view> temp_data;
+    rmm::device_scalar<bool> temp_valid;
+
+    thrust::for_each_n(rmm::exec_policy(stream)->on(stream), 
+                       thrust::make_constant_iterator(0), 1, 
+      [ 
+        buffer=temp_data.data(),
+        validity=temp_valid.data(),
+        d_col=*device_col,
+        index 
+      ] __device__ (auto) mutable {
+        *buffer = d_col.element<string_view>(index);
+        *validity = d_col.is_valid(index);
+      });
+
+    return std::make_unique<string_scalar>(
+      temp_data, temp_valid.value(stream), stream, mr);
+  }
+
+  template<typename T>
+  std::enable_if_t<std::is_same<T, dictionary32>::value, 
+  std::unique_ptr<scalar>> operator()(
+    column_view const& input,
+    size_type index,
+    cudaStream_t stream = 0,
+    rmm::mr::device_memory_resource *mr = rmm::mr::get_default_resource())
+  {
+    auto dict_view = dictionary_column_view(input);
+    auto key_index_scalar = get_element_functor{}.operator()<int32_t>(
+      dict_view.indices(), index, stream);
+
+    size_type key_index = static_cast<numeric_scalar<int32_t> const*>(key_index_scalar.get())->value(stream);
+    auto result = type_dispatcher(dict_view.keys().type(), get_element_functor{},
+                                  dict_view.keys(), key_index, stream, mr);
+
+
+    auto result_validity = result->validity_data();
+    auto device_col = column_device_view::create(input, stream);
+    
+    thrust::for_each_n(rmm::exec_policy(stream)->on(stream), 
+                       thrust::make_constant_iterator(0), 1, 
+      [ result_validity, d_col=*device_col, index ] __device__ (auto) mutable {
+        *result_validity = d_col.is_valid(index);
+      });
+
+    return result;
+  }
+};
+
+} // namespace anonymous
+
+
+std::unique_ptr<scalar> get_element(
+  column_view const& input,
+  size_type index,
+  cudaStream_t stream,
+  rmm::mr::device_memory_resource *mr)
+{
+  CUDF_EXPECTS(index >= 0 and index < input.size(), "Index out of bounds");
```

Comment:
Is this expectation tested in get_value_tests?

Reply:
added unit tests.